src/magellan_motion/src/path_planner/planner.cpp

```diff
@@ -29,24 +31,43 @@ PathPlanner::PathPlanner(ros::NodeHandle& nh, double resolution)
           open_(comp_),
           map_sub(nh.subscribe("/grid", 10, &PathPlanner::mapCallback, this))
 {
-    mapSize = 10 / resolution;
+    nh.getParam("~float_precision", floatPrecision);
+    nh.getParam("~float_precision_divide", floatPrecisionDivide);
+    nh.getParam("~transform_frame", transformFrame);
+
     _has_map = false;
 }
 
-int PathPlanner::getKey(double x, double y) {
+int PathPlanner::getKey(int x, int y) {
     // gets a unique number for any 2 integers
     // first convert to cell coordinates
-    int xMap = (int) x / _resolution;
-    int yMap = (int) y / _resolution;
-    return ((xMap + yMap) * (xMap + yMap + 1) / 2 + yMap);
+    return ((x + y) * (x + y + 1) / 2 + y);
 }
 
-Path PathPlanner::getPlan(std::shared_ptr<Successor> goalNode) {
+Path PathPlanner::getPlan(std::shared_ptr<Successor> goalNode, tf2_ros::Buffer& tfBuffer) {
     Path p;
-    p.header.frame_id = "odom";
+    p.header.frame_id = transformFrame;
     p.header.stamp = ros::Time::now();
     double totalCost = 0;
 
+    geometry_msgs::PoseStamped startPoint;
+    geometry_msgs::PoseStamped startPointTransformed;
+
+    try{
+      auto transform = tfBuffer.lookupTransform(transformFrame,
+                                                "base_link",
+                                                ros::Time::now(),
+                                                ros::Duration(1.0));
+     
+      tf2::doTransform(startPoint, startPointTransformed, transform);
+    } catch (tf2::TransformException &ex) {
+        ROS_ERROR("PathPlanner getPlan: error in lookupTransform");
+        ROS_ERROR("%s",ex.what());
+        return p;
+    }
+
+    //lookup transform between rosparam frame & baselink
```

Comment:
remove comment

Reply:
done.

Reply:
@nls5260 please dont resolve until it has been fixed